src/loaders/workers.js

```diff
@@ -13,8 +13,14 @@ export default ({ queue }) => {
     )
   );
   const workers = new ConnectionLoader(
-    async ({ provisionerId, workerType, options, filter }) => {
-      const raw = await queue.listWorkers(provisionerId, workerType, options);
+    async ({ provisionerId, workerType, options, filter, isQuarantined }) => {
+      const opts = { ...options };
+
+      if (typeof isQuarantined === 'boolean') {
+        opts.quarantined = isQuarantined;
```

Comment:
We can also do this in the method call to avoid copying the object when not needed.

```js
const raw = await queue.listWorkers(
  provisionerId,
  workerType,
  typeof isQuarantined === 'boolean'
    ? { ...options, quarantined: isQuarantined }
    : options,
);
```

Reply:
I like that a lot. Thanks for the insight.